Assets/SEE/Game/SceneManipulation/SpatialMetrics.cs

```diff
@@ -0,0 +1,253 @@
+                throw new IndexOutOfRangeException($"Given direction is not possible in {nameof(Bounds2D)}: {dir}");
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the element at the specified direction.
+        /// </summary>
+        /// <param name="dir">The direction of the element to get or set.</param>
+        /// <returns>The element at the specified direction.</returns>
+        /// <exception cref="IndexOutOfRangeException">
+        /// Thrown when the direction is <see cref="Direction2D.None"/>.
+        /// </exception>
+        public float this[Direction2D dir]
+        {
+            get {
+                switch (dir)
+                {
+                    case Direction2D.Left:
+                        return Left;
+                    case Direction2D.Right:
+                        return Right;
+                    case Direction2D.Back:
+                        return Back;
+                    case Direction2D.Front:
+                        return Front;
+                    default:
+                        throw new IndexOutOfRangeException($"Given direction is not possible in {nameof(Bounds2D)}: {dir}");
+                }
+            }
+
+            set {
+                switch (dir)
+                {
+                    case Direction2D.Left:
+                        Left = value;
+                        break;
+                    case Direction2D.Right:
+                        Right = value;
+                        break;
+                    case Direction2D.Back:
+                        Back = value;
+                        break;
+                    case Direction2D.Front:
+                        Front = value;
+                        break;
+                    default:
+                        throw new IndexOutOfRangeException($"Given direction is not possible in {nameof(Bounds2D)}: {dir}");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Implicit conversion to string.
+        /// <summary>
+        public static implicit operator string(Bounds2D bounds)
+        {
+            return bounds.ToString();
+        }
+
+        /// <summary>
+        /// Returns a printable string with the struct's values.
+        /// <summary>
+        public override readonly string ToString()
+        {
+            return $"{nameof(Bounds2D)}(Left: {Left}, Right: {Right}, Back: {Back}, Front: {Front})";
+        }
+
+        /// <summary>
+        /// Checks if there is an overlap between two bounds.
+        /// </summary>
+        /// <param name="other">The bounds of another object.</param>
+        /// <returns><c>true</c> iff the bounds overlap.</returns>
+        public bool HasOverlap(Bounds2D other)
+        {
+            if (Front < other.Back || Back > other.Front || Left > other.Right || Right < other.Left)
+            {
+                // No overlap detected
+                return false;
+            }
+            return true;
```

Comment:
Can be rewritten as:
```suggestion
            return Front >= other.Back && Back <= other.Front && Left <= other.Right && Right >= other.Left;
```